riscv/csrs.cc

```diff
@@ -544,11 +544,16 @@ mstatus_csr_t::mstatus_csr_t(processor_t* const proc, const reg_t addr):
   val(compute_mstatus_initial_value()) {
 }
 
+reg_t mstatus_csr_t::read() const noexcept {
+  return val & ((state->menvcfg->read() & MENVCFG_DTE) ? ~0ul : ~MSTATUS_SDT);
```

Comment:
This does the wrong thing if `long` is 32 bits, as it is on a 32-bit host.  It'll clear the 32 MSBs.  I'd rewrite it as `return val & ~reg_t(state->menvcfg->read() & MENVCFG_DTE ? 0 : MSTATUS_SDT);`

Reply:
> If all the world was 64bits

That would be nice...